library/std/src/sys/unix/process/process_unix/tests.rs

```diff
@@ -0,0 +1,22 @@
+#[test]
+fn test_command_fork_no_unwind() {
+    use crate::os::unix::process::CommandExt;
+    use crate::os::unix::process::ExitStatusExt;
+    use crate::panic::catch_unwind;
+    use crate::process::Command;
+
+    let got = catch_unwind(|| {
+        let mut c = Command::new("echo");
+        c.arg("hi");
+        unsafe {
+            c.pre_exec(|| panic!("crash now!"));
+        }
+        let st = c.status().expect("get status");
+        eprintln!("{:?}", st);
+        st
+    });
+    eprintln!("got={:?}", &got);
+    let estatus = got.expect("panic not propagated");
+    let signal = estatus.signal().expect("child died of signal");
```

Comment:
`expect()` just prints the string as the panic message when it fails. So this would print

```
thread 'test_command_fork_no_unwind' panicked at 'child died of signal'
```
when the process did *not* exit with a signal.

(Just using `unwrap()` in a test like this is also fine.)

---

The other `.expect()` message also seems inverted.

Reply:
Thanks.  I have fixed the sense of the messages.  Since this test is a bit convoluted (lots stacked up negatives, because it's testing error cases) I think the `expect` messages are useful here.  But they ought to have the right sense indeed!